internal/scrapertest/util.go

```diff
@@ -54,3 +54,26 @@ func sortInstrumentationLibrary(a, b pmetric.ScopeMetrics) bool {
 	}
 	return false
 }
+
+func sortResourceMetrics(a, b pmetric.ResourceMetrics) bool {
+	if a.SchemaUrl() < b.SchemaUrl() {
+		return true
+	}
+	if a.ScopeMetrics().Len() < b.ScopeMetrics().Len() {
+		return true
+	}
+	if a.ScopeMetrics().Len() == b.ScopeMetrics().Len() {
+		for i := 0; i < a.ScopeMetrics().Len(); i++ {
+			aSm := a.ScopeMetrics().At(i)
+			bSm := b.ScopeMetrics().At(i)
+			if aSm.Metrics().Len() < bSm.Metrics().Len() {
+				return true
+			}
+		}
+	}
+	return false
```

Comment:
This is basically handling the case of `a.ScopeMetrics().Len() > b.ScopeMetrics().Len()`.

Let's group this with the opposite case and handle them together:
```go
	if a.ScopeMetrics().Len() != b.ScopeMetrics().Len() {
		return a.ScopeMetrics().Len() < b.ScopeMetrics().Len()
	}
	// now we do not need this next condition either
	// if a.ScopeMetrics().Len() == b.ScopeMetrics().Len() {
```

Reply:
ok

Reply:
> we do not need this next condition either

Sorry @Frapschen, I only meant to remove the condition. The comparison that was within the condition is still valid.

Reply:
@djaglowski fix

Reply:
@djaglowski fix